### PR TITLE
chore: refactor test contracts to fixtures

### PR DESCRIPTION
**Pull request by tserg**
### What I did

Convert the following types of test contracts in the test suite to fixtures:
- does not take in parametrized values and are compiled without any changes for each parametrized value in the test function;
```
@pytest.mark.parametrize(...)
def test_foo():
  code = """
  ...
  """
  c = get_contract(code) # <-- gets run N times
  # do something with c
```
- takes in parametrized values, but the parametrized values are not also used in the test function, and the test function contains other parametrized values
```
@pytest.mark.parametrize(a, [...])
@pytest.mark.parametrize(b, [...])
def test_foo():
  code = f"""
  ...{a}
  """
  c = get_contract(code) # <-- gets run N times
  # do something with c and b, but not a
```

### How I did it

Extract the test contract as a fixture, and pass the fixture instead to the test function.

### How to verify it

Run tests.

### Description for the changelog

Refactor test contracts to fixtures

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by charles-cooper** [comment]
note: to compare timings between https://github.com/vyperlang/vyper/actions/runs/2042620929 vs https://github.com/vyperlang/vyper/actions/runs/2044843484

**Comment by charles-cooper** [comment]
looks like each test comes down by an average of 15 seconds

File: tests/parser/features/test_clampers.py [hunk 1]
```diff
@@ -205,108 +201,166 @@ def foo(s: decimal) -> decimal:
...
     values = [0] * 12
     values[idx] = bad_value
 
-    c = get_contract(code)
-    assert_tx_failed(lambda: _make_tx(w3, c.address, "foo(int128[6][1][2]])", values))
+    assert_tx_failed(
+        lambda: _make_tx(
+            w3,
+            multidimension_array_clamper_failing.address,
+            "foo(int128[6][1][2]])",
```
Comment (ru0xjr669) [inline tests/parser/features/test_clampers.py:363]: Sudo

**Comment by charles-cooper** [comment]
closing on account of the merge conflicts. we can return to this on a case-by-case basis later

**Pull request closed without merging**